iroh-resolver/src/balanced_tree.rs

```diff
@@ -234,13 +234,16 @@ impl TreeNode {
 #[cfg(test)]
 mod tests {
     use super::*;
+    use bytes::BytesMut;
     use futures::StreamExt;
 
     // chunks are just a single usize integer
     const CHUNK_SIZE: u64 = std::mem::size_of::<usize>() as u64;
 
-    fn test_chunk_stream(num_chunks: usize) -> impl Stream<Item = std::io::Result<BytesMut>> {
-        futures::stream::iter((0..num_chunks).map(|n| Ok(BytesMut::from(&n.to_be_bytes()[..]))))
+    fn test_chunk_stream(num_chunks: usize) -> impl Stream<Item = std::io::Result<Bytes>> {
+        futures::stream::iter(
+            (0..num_chunks).map(|n| Ok(BytesMut::from(&n.to_be_bytes()[..]).freeze())),
```

Comment:
I think `n.to_be_bytes().to_vec().into()` looks nicer. It should have the same number of allocations.